src/noit_metric_tag_search.c

```diff
@@ -200,8 +243,13 @@ var_graphite_match(void *impl_data, const char *pattern, const char *in, size_t
   /* If the fixed_prefix len is the subject len, we know we have an exact match */
   if(g->fixed_prefix_len == in_len) return mtev_true;
   for(int i=0; i<g->count; i++) {
-    if(g->results[i].re) {
-      rv = pcre_exec(g->results[i].re, g->results[i].re_e, in, in_len, 0, 0, ovector, 30);
+    if(g->results[i].rem && g->results[i].rem->re == NULL) {
+      re_matcher_compile(g->results[i].rem);
+    }
+    if(g->results[i].rem && g->results[i].rem->re) {
+      ck_pr_faa_64(&g->results[i].rem->execution_counter, 1);
```

Comment:
```suggestion
      g->results[i].rem->execution_counter++;
```

Reply:
The cache is thread local, the rem is not guaranteed to be accessed/executed from a single thread.